### PR TITLE
Implement msg.sig

### LINKED ISSUE #278: Implement msg.sig
### What is wrong?

We do not yet implement reading `msg.sig`

https://github.com/ethereum/fe/blob/177ee35ba4b12150f5cb14152e066d364b1a0161/compiler/src/yul/mappers/expressions.rs#L440

### How can it be fixed

Start here:

https://github.com/ethereum/fe/blob/177ee35ba4b12150f5cb14152e066d364b1a0161/compiler/src/yul/mappers/expressions.rs#L440

Background:

https://docs.soliditylang.org/en/develop/units-and-global-variables.html#block-and-transaction-properties

### PR DESCRIPTION
**Pull request by georgeroman**
Fixes #278 

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [ ] OPTIONAL: Update [Spec](https://github.com/ethereum/fe/blob/master/spec/index.md) if applicable
- [ ] Add entry to the [release notes](https://github.com/ethereum/fe/blob/master/newsfragments/README.md) (may forgo for trivial changes)

- [ ] Clean up commit history

### CODE REVIEW

File: compiler/tests/fixtures/features/return_msg_sig.fe [hunk 1]
```diff
@@ -0,0 +1,5 @@
+contract Foo:
+    pub def bar() -> bytes[32]:
+        msgSig: bytes[32]
```
Comment (cburgdorf) [inline compiler/tests/fixtures/features/return_msg_sig.fe:3]: The idiomatic way to write Fe code is with `snake_case`. Please change `msgSig` to `msg_sig`.

File: compiler/tests/utils.rs [hunk 2]
```diff
@@ -316,6 +317,19 @@ pub fn bytes_token(s: &str) -> ethabi::Token {
     ethabi::Token::FixedBytes(ethabi::FixedBytes::from(s))
 }
 
+#[allow(dead_code)]
+pub fn bytes32(s: &str) -> Vec<u8> {
+    H256::from_str(s)
```
Comment (cburgdorf) [inline compiler/tests/utils.rs:322]: Just pointing out that we try to not use single letter variable names such as `s`. That said, this file violates that rule numerous times and we don't need to fix that in this PR :)

**Review by cburgdorf — APPROVED** [review]
Thanks for the PR. Looks good. Just a tiny nitpick about the casing. Also do you mind creating a file `311.feature.md` in https://github.com/ethereum/fe/tree/master/newsfragments with the content being something like:

>Support for `msg.sig` to read the function identifier

This is to ensure the feature gets included in the next release notes.

**Comment by georgeroman** [comment]
Addressed all comments.

**Comment by cburgdorf** [comment]
Awesome! Landed :tada:

**Pull request merged**